### ISSUE TITLE
Sage 9.3

### ISSUE THREAD

**Issue by edgarcosta** [LMFDB/lmfdb#4596]
LMFDB makes Sage 9.3 segfault on OSX when calling sage.libs.pari.convert_gmp.new_gen_from_mpz_t, i.e., converting from pari's data type to gmp's data type at the C level. Here is a traceback: https://gist.github.com/alexjbest/4058fcc8acee924142b46540bbe2c33d

@roed314 and @alexjbest observe the exact same issue.

Has anyone else observed this before?
@roed314 and I are trying to figure out the reason, but we are a bit afraid that it has to do with threading from werkzeurg and pari not being built threadsafe

**Comment by edgarcosta** [comment]
Here is perhaps a minimal example:
```
from concurrent.futures import ThreadPoolExecutor
with ThreadPoolExecutor() as e:
    j = e.submit(is_squarefree, 15)
j.result()
```

**Comment by roed314** [comment]
I asked on sage-devel: https://groups.google.com/g/sage-devel/c/NmYojQqWzII

**Comment by roed314** [comment]
And we've created a [cypari issue](https://github.com/sagemath/cypari2/issues/107).

**Comment by roed314** [comment]
Fixed by #4603
